### PR TITLE
Adding support for sign request

### PR DESCRIPTION
**Pull request by mhagmajer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,6 +113,7 @@ Command Topics
 * [`box retention-policies`](docs/retention-policies.md) - List all retention policies for your enterprise
 * [`box search`](docs/search.md) - Search for files and folders in your Enterprise
 * [`box shared-links`](docs/shared-links.md) - Manage shared links
+* [`box sign-requests`](docs/sign-requests.md) - Manage sign requests
```
Comment (mwwoda) [inline README.md:116]: This file is missing.

File: test/fixtures/sign-requests/post_sign_requests_id_resend.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (mwwoda) [inline test/fixtures/sign-requests/post_sign_requests_id_resend.json:1]: Do we need this file? Can the test be written without it (resend endpoint returns only status code)

File: src/commands/sign-requests/create.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+class SignRequestsCreateCommand extends BoxCommand {
+	async run() {
+		const { flags, args } = this.parse(SignRequestsCreateCommand);
+
+		const signRequest = await this.client.signRequests.create({
+			signers: flags.signers.split(',').map((email) => ({
+				role: 'signer',
+				email,
+			})),
```
Comment (mwwoda) [inline src/commands/sign-requests/create.js:14]: We don't map any other signers fields (other roles, order etc.)? You think that they will not be used in the CLI?

File: src/commands/sign-requests/create.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const BoxCommand = require('../../box-command');
+const { flags } = require('@oclif/command');
+
+class SignRequestsCreateCommand extends BoxCommand {
+	async run() {
+		const { flags, args } = this.parse(SignRequestsCreateCommand);
+
+		const signRequest = await this.client.signRequests.create({
```
Comment (mwwoda) [inline src/commands/sign-requests/create.js:10]: We don't map any other sign request fields (email_message, prefill_tags etc.)? You think that they will not be used in the CLI?

File: package-lock.json [hunk 5]
```diff
@@ -794,7 +794,7 @@
     },
     "ansi-escapes": {
       "version": "3.1.0",
-      "resolved": "https://registry.npmjs.org/ansi-escapes/-/ansi-escapes-3.1.0.tgz",
+      "resolved": "http://registry.npmjs.org/ansi-escapes/-/ansi-escapes-3.1.0.tgz",
```
Comment (mwwoda) [inline package-lock.json:797]: Just a bit strange that it changed from https to http. Is there a reason for that?

**Review by mwwoda — COMMENTED** [review]
Please also update CHANGELOG.md

File: package-lock.json (hunk 5 above)
Comment (mhagmajer) [reply package-lock.json:797]: Not sure. Didn't make this change manually - `npm install` did. I'm on version `6.14.12`

File: README.md (hunk 1 above)
Comment (mhagmajer) [reply README.md:116]: I think these are automatically from the source code on release - take a look at commit history https://github.com/box/boxcli/tree/main/docs

File: src/commands/sign-requests/create.js [hunk 6]
```diff
@@ -1,56 +1,196 @@
...
+				case 'order':
+					signer.order = value;
+					break;
+
+				case 'embed_url_external_user_id':
+					signer.embed_url_external_user_id = value;
+					break;
+
+				default:
+					throw new BoxCLIError(`Unknown proprty for signer: ${key}`);
```
Comment (mwwoda) [inline src/commands/sign-requests/create.js:97]: typo, should be 'property'

File: src/commands/sign-requests/create.js [hunk 7]
```diff
@@ -1,56 +1,196 @@
...
+					}
+					prefillTag.checkbox_value = value === '1';
+					break;
+
+				case 'date':
+					prefillTag.date_value = value;
+					break;
+
+				default:
+					throw new BoxCLIError(`Unknown proprty for prefill-tag: ${key}`);
```
Comment (mwwoda) [inline src/commands/sign-requests/create.js:180]: typo, should be property

File: src/commands/sign-requests/create.js [hunk 8]
```diff
@@ -1,56 +1,196 @@
...
 
 		await this.output(signRequest);
 	}
 }
 
 SignRequestsCreateCommand.description = 'Create sign request';
-SignRequestsCreateCommand.examples = [
-	'box sign-requests:create -S alice@example.com -f 12345 -p 23456',
-];
+SignRequestsCreateCommand.examples = ['box sign-requests:create --signer email=alice@example.com --source-files 12345 --parent-folder 23456  --prefil-tag id=1,value=Test'];
```
Comment (mwwoda) [inline src/commands/sign-requests/create.js:42]: typo, should be 'prefill'

File: src/commands/sign-requests/create.js [hunk 9]
```diff
@@ -1,56 +1,196 @@
...
+	'email-message': flags.string({
+		description:
+			'Message to include in sign request email. The field is cleaned through sanitization of specific characters. However, some html tags are allowed. Links included in the message are also converted to hyperlinks in the email. The message may contain the following html tags including `a`, `abbr`, `acronym`, `b`, `blockquote`, `code`, `em`, `i`, `ul`, `li`, `ol`, and `strong`. Be aware that when the text to html ratio is too high, the email may end up in spam filters. Custom styles on these tags are not allowed. If this field is not passed, a default message will be used.',
+	}),
+	'reminders-enabled': flags.boolean({
+		description:
+			'Reminds signers to sign a document on day 3, 8, 13 and 18. Reminders are only sent to outstanding signers.',
+		allowNo: true,
+	}),
+	'prefill-tag': flags.string({
```
Comment (mwwoda) [inline src/commands/sign-requests/create.js:144]: It should be plural - 'tags', right?

File: src/commands/sign-requests/create.js (hunk 8 above)
Comment (mwwoda) [inline src/commands/sign-requests/create.js:42]: It should be plural - 'tags', right?

**Review by mwwoda — APPROVED** [review]
Approving with minor comments (typos). Overall, that's a pretty neat way to handle construction of such a complex object in the CLI!

File: src/commands/sign-requests/create.js (hunk 9 above)
Comment (mhagmajer) [reply src/commands/sign-requests/create.js:144]: I made it singular because using this flag you can set one tag. The flag can be used multiple times.

**Pull request merged**